### PR TITLE
go/runtime: Reduce downtime for TEE runtime upgrades

### PR DESCRIPTION
**Pull request by kostko**
Based on #5187

### CODE REVIEW

File: go/worker/common/committee/node.go [hunk 1]
```diff
@@ -432,10 +433,21 @@ func (n *Node) updateHostedRuntimeVersionLocked() {
 		activeVersion = activeDeploy.Version
 	}
 
-	if err := n.SetHostedRuntimeVersion(n.ctx, activeVersion); err != nil {
-		n.logger.Error("failed to activate runtime version",
+	// For compute nodes, determine if there is a next version and activate it early.
+	var nextVersion *version.Version
+	if config.GlobalConfig.Mode == config.ModeCompute {
+		nextDeploy := n.CurrentDescriptor.NextDeployment(n.CurrentEpoch)
+		if nextDeploy != nil {
```
Comment (peternose) [inline go/worker/common/committee/node.go:440]:
```suggestion
		if nextDeploy != nil && nextDeploy.ValidFrom - n.CurrentEpoch < 12345 {
```
We probably don't want to start doing attestations to early.

**Review by peternose — CHANGES_REQUESTED** [review]
Tested locally and warm-up works. Nice :+1:

File: go/runtime/host/sandbox/sandbox.go [hunk 2]
```diff
@@ -225,6 +241,13 @@ func (r *sandboxedRuntime) Stop() {
 
 // Implements host.EmitEvent.
 func (r *sandboxedRuntime) EmitEvent(ev *host.Event) {
+	// Update runtime's CapabilityTEE in case this is an update event.
```
[the remark this answers is not in the document]
Comment (kostko) [reply go/runtime/host/sandbox/sandbox.go]: Agreed, refactored this a bit.

File: go/worker/common/committee/node.go (hunk 1 above)
Comment (kostko) [reply go/worker/common/committee/node.go]: Agreed, made this configurable, please take a look.

**Pull request merged**